lib/api/apiUtils/integrity/validateChecksums.js

```
@@ -0,0 +1,27 @@
+const crypto = require('crypto');
+
+const ChecksumError = Object.freeze({
+    MD5Mismatch: 'MD5Mismatch',
+    MissingChecksum: 'MissingChecksum',
+});
+
+/**
+ * validateChecksumsNoChunking - Validate the checksums of a request.
+ * @param {object} headers - http headers
+ * @param {string} body - http request body
+ * @return {string} - error message
+ */
+function validateChecksumsNoChunking(headers, body) {
+    if (headers && headers['content-md5']) {
+        const md5 = crypto.createHash('md5').update(body, 'utf8').digest('base64');
+        if (md5 !== headers['content-md5']) {
+            return { error: ChecksumError.MD5Mismatch, details: { calculated: md5, expected: headers['content-md5'] } };
+        }
```

We currently do nothing of this object, when we handle errors. Should we just return the error and add a log in this function?

---

I think the caller should do the logging.

In the next iteration I will add `SHA256Mismatch`, `CRC32Mismatch`, etc. I don't see why we would want to squash all the possible errors into `BadDigest`